src/node/historical_queries_interface.h

```diff
@@ -27,4 +28,108 @@ namespace ccf::historical
       return nullptr;
     }
   };
+
+  using CheckAvailability = std::function<bool(
+    kv::Consensus::View view,
+    kv::Consensus::SeqNo seqno,
+    std::string& error_reason)>;
+
+  using HandleHistoricalQuery = std::function<void(
+    ccf::RequestArgs& args,
+    StorePtr store,
+    kv::Consensus::View view,
+    kv::Consensus::SeqNo seqno)>;
+
+  static ccf::HandleFunction adapter(
+    const HandleHistoricalQuery& f,
+    AbstractStateCache& state_cache,
+    const CheckAvailability& available)
+  {
+    return [f, &state_cache, available](RequestArgs& args) {
+      // Extract the requested transaction ID
+      kv::Consensus::View target_view;
+      kv::Consensus::SeqNo target_seqno;
+
+      {
+        const auto target_view_opt =
```

Comment:
Why are we using HTTP headers instead of query string here? Is that because the view and seqno are already returned as response headers when a transaction is committed. In other words, the view and seqno are related CCF whereas the everything app-related uses query strings?

Reply:
I'm fine with putting view and seqno in the query string/body in other places - that's what we do with `getReceipt`, getTxStatus` etc.

But in this case I'm trying to make a distinction between the things the framework/wrapper does, and what the app does. We could parse a few things from the query string and leave the rest for the app but this seems ugly - we don't currently split the query for easy consumption _in the framework_ at all, we're deliberately leaving that for the app to decide what types it expects. I think its cleaner if this framework-defined wrapper uses framework-defined headers, and still leaves the query/body parsing to the app.

Also just a note, not sure if this PR makes it clear, this adapter is entirely optional - its still possible to write a historical query entirely in the app that use your own headers, or the query string, or whatever you like.

Reply:
I agree with you - it is much better/cleaner this way. The only minor drawback I can think of is that it's _slightly_ more fiddly to set a HTTP request header than a query string with `curl`.